Source/Core/Gen4/Encounters4.cpp

```diff
@@ -272,6 +272,125 @@ namespace Encounters4
+                                    level = entry[(9 * modifier) + (waterBlockSlot * 3) + 314];
+                                    waterBlockSlot++;
+                                    break;
+                                }
+
+                                waterBlockSlot++;
+                            }
+                        }
+                        else if (waterFlag != 0 && safariEncounter == Encounter::OldRod)
+                        {
+                            specie = *reinterpret_cast<const u16 *>(entry + (30 * modifier) + (i * 3) + 351);
+                            level = entry[(30 * modifier) + (i * 3) + 353];
+
+                            while (waterBlockSlot < 2)
+                            {
+                                if (blocks[entry[459 + (waterBlockSlot * 4)]] >= entry[460 + (waterBlockSlot * 4)] &&
+                                        blocks[entry[461 + (waterBlockSlot * 4)]] >= entry[462 + (waterBlockSlot * 4)])
+                                {
+                                    specie = *reinterpret_cast<const u16 *>(entry + (6 * modifier) + (waterBlockSlot * 3) + 441);
+                                    level = entry[(6 * modifier) + (waterBlockSlot * 3) + 443];
+                                    waterBlockSlot++;
+                                    break;
+                                }
+
+                                waterBlockSlot++;
+                            }
+                        }
+                        else if (waterFlag != 0 && safariEncounter == Encounter::GoodRod)
+                        {
+                            specie = *reinterpret_cast<const u16 *>(entry + (30 * modifier) + (i * 3) + 467);
+                            level = entry[(30 * modifier) + (i * 3) + 469];
+
+                            while (waterBlockSlot < 2)
+                            {
+                                if (blocks[entry[575 + (waterBlockSlot * 4)]] >= entry[576 + (waterBlockSlot * 4)] &&
+                                        blocks[entry[577 + (waterBlockSlot * 4)]] >= entry[578 + (waterBlockSlot * 4)])
+                                {
+                                    specie = *reinterpret_cast<const u16 *>(entry + (6 * modifier) + (waterBlockSlot * 3) + 557);
+                                    level = entry[(6 * modifier) + (waterBlockSlot * 3) + 559];
+                                    waterBlockSlot++;
+                                    break;
+                                }
+
+                                waterBlockSlot++;
+                            }
+                        }
+                        else if (waterFlag != 0 && safariEncounter == Encounter::SuperRod)
+                        {
+                            specie = *reinterpret_cast<const u16 *>(entry + (30 * modifier) + (i * 3) + 583);
+                            level = entry[(30 * modifier) + (i * 3) + 585];
+
+                            while (waterBlockSlot < 2)
```

Comment:
for (int waterBlockSlot = 0; waterBlockSlot < 2; waterBlockSlot) {

}

Reply:
This makes things not working